File: 0001-0500/0007/0007.reverse-integer.py
#
# @lc app=leetcode id=7 lang=python3
#
# [7] Reverse Integer
#

# @lc code=start
class Solution:
    def reverse(self, x: int) -> int:
        res = 0
        if x >= 0:
            res= int(str(x)[::-1])
        else:
            res = int(str(x)[::-1][:-1])*-1
        return res if (-2**31 <= res and res <= 2**31 - 1) else 0
# @lc code=end

File: 0001-0500/0007/0007.reverse-integer.js
/*
 * @lc app=leetcode id=7 lang=javascript
 *
 * [7] Reverse Integer
 */

// @lc code=start
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
    let result = 0;
    while (x !== 0) {
        result = result * 10 + x % 10;
        x = x / 10 | 0;
    }
    return (result | 0) === result ? result : 0;
};
// @lc code=end

